akka-actor-tests/src/test/scala/akka/actor/ActorPathSpec.scala

```diff
@@ -89,5 +89,15 @@ class ActorPathSpec extends WordSpec with Matchers {
       ActorPath.fromString("akka://mysys/user/boom/*")
     }
 
+    "detect invalid chars in host names when not using AddressFromURIString, e.g. docker host given name" in {
+      Address("akka", "sys", Some("valid"), Some(0)).hasInvalidHostCharacters shouldBe false
+      Address("akka", "sys", Some("in_valid"), Some(0)).hasInvalidHostCharacters shouldBe true
+      intercept[MalformedURLException](AddressFromURIString("akka://sys@in_valid:5001"))
+    }
```

Comment:
AFAIK underscore is valid in domain names though, so cover that as well in case the hostname is a full domain name, for example: "valid._local.org"

Probably not that common but annoying if we block something that is valid.

Reply:
👍 full domain names with underscores are seen in the wild, and yet:

Part of the problem:
Spark 
* https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala#L54

Docker:
* https://github.com/docker/compose/issues/229
* https://github.com/docker/compose/issues/4475
* https://github.com/docker/compose/issues/3912
* https://forums.docker.com/t/underscore-in-domain-names/12584

Reply:
Still, our joins etc (things outside cluster) should fail fast ideally vs pass the buck to the user to validate when it typically comes from config and can not be. I don't think we should force users to dig through Akka async log output for this. I'll upgrade the string.contains to a regex and document it.

Reply:
@johanandren I pushed an update with regex and test updates.